sds/src/scanner/suppression.rs

```diff
@@ -0,0 +1,68 @@
+use ahash::AHashSet;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct SuppressionConfig {
+    #[serde(default)]
+    pub starts_with: Vec<String>,
+    #[serde(default)]
+    pub ends_with: Vec<String>,
+    #[serde(default)]
+    pub exact_match: Vec<String>,
+}
+
+pub struct CompiledSuppressionConfig {
+    pub starts_with: Vec<String>,
+    pub ends_with: Vec<String>,
+    pub exact_match: AHashSet<String>,
+}
+
+impl CompiledSuppressionConfig {
+    pub fn should_match_be_suppressed(&self, match_content: &str) -> bool {
+        if self.starts_with.is_empty() && self.ends_with.is_empty() && self.exact_match.is_empty() {
```

Comment:
This check isn't really needed. The logic below will be fast in this case and return the same thing.